python/semantic_kernel/connectors/ai/nvidia/services/nvidia_text_embedding.py

```diff
@@ -0,0 +1,148 @@
+        if not nvidia_settings.embedding_model_id:
+            raise ServiceInitializationError("The NVIDIA embedding model ID is required.")
+        if not (api_key or nvidia_settings.api_key):
+            raise ServiceInitializationError("Please provide an api_key")
+        if not client:
+            client = AsyncOpenAI(api_key=nvidia_settings.api_key.get_secret_value(), base_url=nvidia_settings.base_url)
+        super().__init__(
+            ai_model_id=nvidia_settings.embedding_model_id,
+            api_key=nvidia_settings.api_key.get_secret_value() if nvidia_settings.api_key else None,
+            ai_model_type=NvidiaModelTypes.EMBEDDING,
+            service_id=service_id or nvidia_settings.embedding_model_id,
+            env_file_path=env_file_path,
+            client=client,
+        )
+
+    @override
+    async def generate_embeddings(
+        self,
+        texts: list[str],
+        settings: "PromptExecutionSettings | None" = None,
+        batch_size: int | None = None,
+        **kwargs: Any,
+    ) -> ndarray:
+        raw_embeddings = await self.generate_raw_embeddings(texts, settings, batch_size, **kwargs)
+        return array(raw_embeddings)
+
+    @override
+    async def generate_raw_embeddings(
+        self,
+        texts: list[str],
+        settings: "NVIDIAEmbeddingPromptExecutionSettings | None" = None,
+        batch_size: int | None = None,
+        **kwargs: Any,
+    ) -> Any:
+        """Returns embeddings for the given texts in the unedited format.
+
+        Args:
+            texts (List[str]): The texts to generate embeddings for.
+            settings (NVIDIAEmbeddingPromptExecutionSettings): The settings to use for the request.
+            batch_size (int): The batch size to use for the request.
+            kwargs (Dict[str, Any]): Additional arguments to pass to the request.
+        """
+        if not settings:
+            settings = NVIDIAEmbeddingPromptExecutionSettings(ai_model_id=self.ai_model_id)
+        else:
+            if not isinstance(settings, NVIDIAEmbeddingPromptExecutionSettings):
+                settings = self.get_prompt_execution_settings_from_settings(settings)
+        assert isinstance(settings, NVIDIAEmbeddingPromptExecutionSettings)  # nosec
+        if settings.model is None:
+            settings.model = self.ai_model_id
+        for key, value in kwargs.items():
+            setattr(settings, key, value)
+
+        # move input_type to extra-body
+        if not settings.extra_body:
+            settings.extra_body = {}
+        settings.extra_body.setdefault("input_type", settings.input_type)
+        raw_embeddings = []
+        batch_size = batch_size or len(texts)
+        for i in range(0, len(texts), batch_size):
+            batch = texts[i : i + batch_size]
+            settings.input = batch
+            raw_embedding = await self._send_request(settings=settings)
```

Comment:
it looks like batch_size is controlling the number of texts that are sent for embedding in a single call to the service.

each of those batches can also be sent in parallel.

Reply:
indeed, batch size refers to how many characters are included in each call, if this can be parallelized, please do, also moving the actual embedding calls here makes this simpler (vs the handler, see earlier comment)